src/chttpd_cors.erl

```diff
@@ -159,17 +159,24 @@ handle_preflight_request(Req, Config, Origin) ->
 
 
 headers(Req, RequestHeaders) ->
-    case get_origin(Req) of
-        undefined ->
-            %% If the Origin header is not present terminate
-            %% this set of steps. The request is outside the scope
-            %% of this specification.
-            %% http://www.w3.org/TR/cors/#resource-processing-model
-            RequestHeaders;
-        Origin ->
-            headers(Req, RequestHeaders, Origin, get_cors_config(Req))
+    case is_preflight_response(RequestHeaders) of
```

Comment:
@chewbranca: `is_preflight_response` is removed in favor of explicit `chttpd:send_response_no_cors/4` function.